saas_test/tests/test_saas_template.py

```diff
@@ -1,34 +1,71 @@
+    with odoo.api.Environment.manage(), db.cursor() as cr:
+        try:
+            _logger.debug('%s: %s', db_name, module_name)
+            env = odoo.api.Environment(cr, SUPERUSER_ID, {})
+            return getattr(env['ir.module.module'], 'search')([('name', '=', module_name), ('state', '=', 'installed')])
+        except:
+            _logger.error('Error while searching module %s in %s', module_name, db_name)
 
 
 @tagged('post_install', 'at_install')
 class TestSaasTemplate(TransactionCase):
+    # TODO: add second optional argument - method_name
+    def perform_last_job(self, model_name):
+        job_id = self.env['queue.job'].search([('model_name', '=', model_name)])
+        if job_id:
+            job = Job.load(self.env, job_id[-1].uuid)
+            if job.state == ENQUEUED or job.state == PENDING:
+                job.perform()
+
     def setUp(self):
         super(TestSaasTemplate, self).setUp()
         self.saas_template = self.env['saas.template'].create({
             'template_modules_domain': 'mail',
-            'template_post_init': "self.env['mail.message'].create({'message_type': 'notification',}).message_format()"
+            'template_post_init': "self.env['mail.message'].create({'message_type': 'notification',}).message_format()",
+        })
+        self.saas_operator = self.env['saas.operator'].create({
+            'type': 'local',
         })
-        self.saas_operator = self.env['saas.operator'].create({})
         self.saas_template_operator = self.env['saas.template.operator'].create({
             'template_id': self.saas_template.id,
             'operator_id': self.saas_operator.id,
-            'operator_db_name': 'test_db_name',
-            'to_rebuild': True,
-            'state': 'draft'
+            'operator_db_name': DB_TEMPLATE,
         })
 
     def test_template_operator(self):
+        # FIXME: that check needed when last tests didn't pass, not sure that it is correct way to drop db
+        if DB_TEMPLATE in db.list_dbs():
+            db.exp_drop(DB_TEMPLATE)
         self.saas_template_operator.preparing_template_next()
-        self.saas_template_operator.operator_db_id.create_db(
-            None,
-            self.saas_template_operator.template_id.template_demo
-        )
+        self.perform_last_job('saas.db')
+        self.perform_last_job('saas.template.operator')
+        self.perform_last_job('saas.template.operator')
+
+
+        # Tests that template db created correctly
         self.assertTrue(self.saas_template_operator.operator_db_id.name)
-        self.assertEqual(self.saas_template_operator.operator_db_id.name, 'test_db_name')
-        self.assertIn('test_db_name', db.list_dbs())
+        self.assertEqual(self.saas_template_operator.operator_db_id.name, DB_TEMPLATE)
+        self.assertIn(DB_TEMPLATE, db.list_dbs())
+
+        # Check that module from template_modules_domain is installed
+        self.assertTrue(is_module_installed(DB_TEMPLATE, 'mail'))
```

Comment:
You can use some assert method inside is_module_installed, which can be renamed to assert_module_is_installed